File: paleale/src/service/beer/protocols.ts
import {List, Map} from "immutable";

import {BeerID} from "../../id";
import {Beer} from "../../model/beer";

export interface AddBeerParams {
  beer: Beer;
}

export interface AddBeerResponse {
  beerID: BeerID;
}

export interface BulkGetBeerByIDsParams {
  beerIDs: List<BeerID>;
}

export interface BulkGetBeerByIDsResponse {
  beers: Map<BeerID, Beer>;
}

export interface SearchBeersDBParams {
  phrase: string;
}

export interface BeerIDAndModel {
  id: string;
  model: Beer;
}

export interface SearchBeersDBResponse {
  beers: List<BeerIDAndModel>;
}

export interface SearchBeersUntappdParams {
  phrase: string;
}

export interface SearchBeersUntappdResponse {
  beers: List<BeerIDAndModel>;
}

File: kloster/modules/beer/protocols.go
package beer

import (
	"bitbucket.org/ctiu6442/m14b-group-1-cp55/kloster/id"
	"bitbucket.org/ctiu6442/m14b-group-1-cp55/kloster/model"
)

type addBeerParams struct {
	Beer *model.Beer `json:"beer"`
}

type addBeerResponse struct {
	BeerID id.BeerID `json:"beerID"`
}

type bulkGetBeerByIDsParams struct {
	BeerIDs []id.BeerID `json:"beerIDs"`
}

type bulkGetBeerByIDsResponse struct {
	Beers map[id.BeerID]*model.Beer `json:"beers"`
}

type searchBeersDBParams struct {
	Phrase string `json:"phrase"`
}

type searchBeersDBResponse struct {
	Beers []IDAndModel `json:"beers"`
}

type searchBeersUntappdParams struct {
	Phrase string `json:"phrase"`
}

type searchBeersUntappdResponse struct {
	Beers []IDAndModel `json:"beers"`
}
